sycl/include/sycl/ext/oneapi/experimental/bfloat16_math.hpp

```diff
@@ -46,6 +46,14 @@ template <size_t N> sycl::marray<bool, N> isnan(sycl::marray<bfloat16, N> x) {
   return res;
 }
 
+template <int N> sycl::vec<bool, N> isnan(sycl::vec<bfloat16, N> x) {
+  sycl::vec<bool, N> res;
+  for (size_t i = 0; i < N; i++) {
+    res[i] = isnan(x[i]);
+  }
+  return res;
+}
```

Comment:
Commenting here about `isnan`, but this same comment applies to all the other functions too.

I realize that some of this predates your change, but the API exposed here is quite different from `sycl::nan`.  The API for `sycl::nan` is:

```
bool isnan(float x)
bool isnan(double x)
bool isnan(half x)

template<typename NonScalar>
/*return-type*/ isnan(NonScalar x)
```

See the [SYCL spec](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_math_functions) for more details on the template parameter `NonScalar`.

I would expect the `bfloat16` support to be added in a similar way:

```
bool isnan(bfloat16 x)

// Constrain NonScalar to be either vec<bfloat16, N> or marray<bfloat16 ,N>
template<typename NonScalar>
/*return-type*/ isnan(NonScalar x)
```

Note also that the return type is different from `sycl::isnan`.  The relational operators for `vec` return `vec<int32_t, N>` not `vec<bool, N>`.

Finally, these builtins need to be designed to allow the input to be the swizzled vec type in addition to `vec`.  See the the SYCL spec link above for information about this, but also see KhronosGroup/SYCL-Docs#514 for a new change to the way the builtin functions handle the swizzled vec type.  I believe @steffenlarsen already has this implemented on a branch.

@steffenlarsen may have other comments here as he has been involved with other changes to these builtin functions recently.